core/client-internal/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -72,8 +72,20 @@ public AlluxioBlockStore getAluxioBlockStore() {
    * {@link ClientContext}.
    */
   public void reset() {
-    mFileSystemMasterClientPool.close();
-    mFileSystemMasterClientPool =
-        new FileSystemMasterClientPool(ClientContext.getMasterAddress());
+    if (null != mFileSystemMasterClientPool && isMasterChange()) {
```

Comment:
It seems like this change will cause less connections to be closed, not more - could you explain the issue this is solving?